ecommerce/templates/ecommerce/product_detail.html

```diff
@@ -45,43 +45,88 @@ <h5 class="card-text font-weight-bold mr-4">{% trans 'Price : ' %}{{ object.pric
   </div>
   <div class="row pt-4">
     <div class="col-md-7">
-      <div class="d-flex justify-content-end pb-4">
-        <div class="d-flex align-items-center flex-nowrap">
-          <label class="fs-sm text-muted text-nowrap me-2 d-none d-sm-block" for="sort-reviews">Sort by: </label>
-          <div class="form-element form-select-sm" style="padding-left: 10px; padding-bottom: 10px;">
-            <select class="form-control" id="sort-reviews">
-              <option>Newest</option>
-              <option>Oldest</option>
-            </select>
+      <div class="row pb-3">
+        <div class="col-lg-8 col-md-5">
+          <h3 class="h3 mb-8">All reviews</h3>
+        </div>
+        <div class="d-flex justify-content-end pb-4">
+          <div class="d-flex align-items-center flex-nowrap">
+            <label class="fs-sm text-muted text-nowrap me-2 d-none d-sm-block" for="sort-reviews">Sort by: </label>
+            <div class="form-element form-select-sm" style="padding-left: 10px; padding-bottom: 10px;">
+              <select class="form-control" id="sort-reviews">
+                <option>Newest</option>
+                <option>Oldest</option>
+              </select>
+            </div>
           </div>
         </div>
       </div>
-      <div class="product-review pb-4 mb-4 border-bottom">
-        <div class="d-flex mb-3">
-          <div class="d-flex align-items-center me-4 pe-2">
-            <img class ="rounded-circle" width="50" src="\media\profile_pics\avatar.png" alt="User">
-            <div class="ps-3" style="padding-left: 20px;">
-              <h6 class="fs-sm mb-0">Ngọc Duyên</h6>
-              <span class="fs-ms text-muted">June 28, 2019</span>
-            </div>
-            <div>
-              <div class="star-rating">
-                <i class="star-rating-icon ci-star-filled active"></i>
-                <i class="star-rating-icon ci-star-filled active"></i>
-                <i class="star-rating-icon ci-star-filled active"></i>
-                <i class="star-rating-icon ci-star-filled active"></i>
-                <i class="star-rating-icon ci-star"></i>
+      {% if review.exists %}
+      {% for rv in review %}
+        <div class="product-review pb-4 mb-4 border-bottom">
+          <div class="d-flex mb-3">
+            <div class="d-flex align-items-center me-4 pe-2">
+              <img class ="rounded-circle" width="50" src="\media\profile_pics\avatar.png" alt="User">
+              <div class="ps-3" style="padding-left: 20px;">
+                <h6 class="fs-sm mb-0">{{ rv.user.username }}</h6>
+                <span class="fs-ms text-muted">{{ rv.created }}</span>
               </div>
             </div>
           </div>
+          {% if rv.rate == 5 %}
+          <div style="padding-bottom: 10px;">
+            {% for i in "12345" %}
+            <i class="fas fa-star" style="color: orange;"></i>
+            {% endfor %}
+          </div>
```

Comment:
khỏi check if else đi em, lấy số lần ở mục rate ra luôn
```suggestion
          <div style="padding-bottom: 10px;">
            {% for i in range(rv.rate) %}
            <i class="fas fa-star" style="color: orange;"></i>
            {% endfor %}
          </div>
```